src/shogun/lib/any.h

```diff
@@ -328,6 +345,18 @@ namespace shogun
 		    [](T lhs, T rhs) -> bool { return any_detail::compare(lhs, rhs); });
 	}
 
+	template <class T, class S>
+	void ArrayReference<T, S>::reset(const ArrayReference<T, S>& other)
+	{
+		auto src = *(other.m_ptr);
+		auto len = *(other.m_length);
+		auto& dst = *(this->m_ptr);
+		delete[] dst;
```

Comment:
@karlnapf We've got to resolve this somehow. It has to be the very same allocator that was used to allocate the array. It could be free vs. delete[] or even worse.

Reply:
if it is a shogun array, we cannot rely on SG_MALLOC SG_FREE?